core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsAddressEndpointGroup.java

```diff
@@ -0,0 +1,163 @@
+     * @param defaultPort the port to use when the DNS answer does not contain one. {@code 0} indicates an
+     *     unspecified port, meaning the port will use a protocol-specific well-defined port number
+     *     (e.g., 80, 443).
+     * @param eventLoop the {@link EventLoop} to schedule DNS queries on.
+     */
+    public static DnsAddressEndpointGroup of(String hostname, int defaultPort, EventLoop eventLoop) {
+        return DnsAddressEndpointGroup.of(hostname, defaultPort, eventLoop, Duration.ofSeconds(1));
+    }
+
+    /**
+     * Creates a {@link DnsAddressEndpointGroup}.
+     *
+     * @param hostname the hostname to query DNS queries for.
+     * @param defaultPort the port to use when the DNS answer does not contain one. {@code 0} indicates an
+     *     unspecified port, meaning the port will use a protocol-specific well-defined port number
+     *     (e.g., 80, 443).
+     * @param eventLoop the {@link EventLoop} to schedule DNS queries on.
+     * @param queryInterval the {@link Duration} to query DNS at.
+     */
+    public static DnsAddressEndpointGroup of(String hostname, int defaultPort, EventLoop eventLoop,
+                                             Duration queryInterval) {
+        return new DnsAddressEndpointGroup(hostname, defaultPort,
+                                           DnsEndpointGroupUtil.createResolverForEventLoop(eventLoop),
+                                           eventLoop, queryInterval);
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(DnsAddressEndpointGroup.class);
+
+    private final String hostname;
+    private final int defaultPort;
+    private final NameResolver<InetAddress> resolver;
+    private final EventLoop eventLoop;
+    private final Duration queryInterval;
+
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    @VisibleForTesting
+    DnsAddressEndpointGroup(String hostname, int defaultPort, NameResolver<InetAddress> resolver,
+                            EventLoop eventLoop, Duration queryInterval) {
+        checkArgument(defaultPort >= 0 && defaultPort <= 65535, "defaultPort must be between 0 and 65535");
+        this.hostname = requireNonNull(hostname, "hostname");
+        this.defaultPort = defaultPort;
+        this.resolver = requireNonNull(resolver, "resolver");
+        this.eventLoop = requireNonNull(eventLoop, "eventLoop");
+        this.queryInterval = requireNonNull(queryInterval, "queryInterval");
+    }
+
+    /**
+     * Starts polling for service updates.
+     */
+    public void start() {
+        checkState(scheduledFuture == null, "already started");
+        scheduledFuture = eventLoop.scheduleAtFixedRate(this::query, 0, queryInterval.getSeconds(),
+                                                        TimeUnit.SECONDS);
+    }
+
+    /**
+     * Stops polling for service updates.
+     */
+    @Override
+    public void close() {
+        scheduledFuture.cancel(true);
```

Comment:
Think it would be better to add null check to avoid NullPointerException. (e.g. Create an instance but not started polling)